lochness/email/__init__.py

```diff
@@ -200,6 +205,40 @@ def send_out_daily_updates(Lochness, days: int = 1,
 
         in_mail_footer = 'Only the files transferred to NDA are shown.'
 
+        if shorten:
+            s3_df_tmp = s3_df_selected.copy()
+            for index, row in s3_df_tmp.iterrows():
+                s3_df_tmp.loc[index, 'File name'] = re.sub(
+                        row['Subject'], 'SUBJECT', row['File name'])
+
+                mindlamp_act_patt = r'_activity_(\d{4}_\d{2}_\d{2}.+)$'
+                if re.search(mindlamp_act_patt, row['File name']):
+                    rest_str = re.search(mindlamp_act_patt,
+                                         row['File name']).group(1)
+                    s3_df_tmp.loc[index, 'File name'] = \
+                            f'SUBJECT_activity_{rest_str}'
+
+                for patt in ['-diaryTranscriptQC-day', '_audioJournal_day',
+                        '-diaryAudioQC-day']:
+                    if re.search(patt,
+                                 row['File name']):
+                        s3_df_tmp.loc[index, 'File name'] = \
+                                re.search(r'([a-zA-Z]+)[-_]day', patt).group(1)
+
+            gb = s3_df_tmp.groupby('File name')
+            for unique_filename, table in gb:
+                s3_df_tmp.loc[table.index, 'Subject'] = \
+                    f'{len(table)} subjects'
+                s3_df_tmp.loc[table.index, 'Transfer time (UTC)'] = \
+                        table['Transfer time (UTC)'].iloc[0]
+                s3_df_tmp.loc[table.index, 'Download time (UTC)'] = \
+                        table['Download time (UTC)'].iloc[0]
+            s3_df_selected = s3_df_tmp.drop_duplicates()
+            rename_Subject = lambda x: 'Number of subjects' if x == 'Subject' \
+                    else x
+            s3_df_selected.columns = [rename_Subject(x) for x in
+                                      s3_df_selected.columns]
+
```

Comment:
We are grouping all mindlamp activity and dairy related files transferred, so we get only one row for them. 

This should indeed shorten our emails. Looks good!